src/System.Linq.Parallel/tests/QueryOperators/ForAllTests.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Linq;
+using System.Threading;
+using Xunit;
+
+namespace Test
+{
+    public class ForAllTests
+    {
+        [Theory]
+        [MemberData("Ranges", (object)(new int[] { 1, 2, 16 }), MemberType = typeof(Sources))]
+        [MemberData("Ranges", (object)(new int[] { 1, 2, 16 }), MemberType = typeof(UnorderedSources))]
+        public static void ForAll(Labelled<ParallelQuery<int>> labelled, int count)
+        {
+            ParallelQuery<int> query = labelled.Item;
+            IntegerRangeSet seen = new IntegerRangeSet(0, count);
+            query.ForAll<int>(x => Assert.True(seen.Add(x)));
```

Comment:
@stephentoub  - nope nope nope, locking a BitArray to do this safely dies in the face of outerloop tests.  I think I'm going to need my own version that locks by partition or something.  Using Interlocked.Increment for the counting set also runs into problems at scale (although I use that class less).